torchrl/data/replay_buffers/rb_prototype.py

```diff
@@ -0,0 +1,248 @@
+import collections
+import threading
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any, Callable, Optional, Sequence, Union, Tuple
+
+import torch
+
+from ..tensordict.tensordict import TensorDictBase, LazyStackedTensorDict
+from .replay_buffers import pin_memory_output, stack_tensors, stack_td
+from .samplers import Sampler, RandomSampler
+from .storages import Storage, ListStorage
+from .utils import INT_CLASSES, to_numpy
+from .writers import Writer, RoundRobinWriter
+
+
+class ReplayBuffer:
+    """
+    #TODO: Description of the ReplayBuffer class needed.
+    Args:
+        storage (Storage, optional): the storage to be used. If none is provided
+            a default ListStorage with max_size of 1_000 will be created.
+        sampler (Sampler, optional): the sampler to be used. If none is provided
+            a default RandomSampler() will be used.
+        writer (Writer, optional): the writer to be used. If none is provided
+            a default RoundRobinWriter() will be used.
+        collate_fn (callable, optional): merges a list of samples to form a
+            mini-batch of Tensor(s)/outputs.  Used when using batched
+            loading from a map-style dataset.
+        pin_memory (bool): whether pin_memory() should be called on the rb
+            samples.
+        prefetch (int, optional): number of next batches to be prefetched
+            using multithreading.
+    """
+
+    def __init__(
+        self,
+        storage: Optional[Storage] = None,
+        sampler: Optional[Sampler] = None,
+        writer: Optional[Writer] = None,
+        collate_fn: Optional[Callable] = None,
+        pin_memory: bool = False,
+        prefetch: Optional[int] = None,
+    ) -> None:
+        self._storage = storage or ListStorage(max_size=1_000)
```

Comment:
This do not work. I tried to set storage = LazyMemmapStorage but it got reseted to ListStorage